src/sources/apache_metrics/parser.rs

```diff
@@ -0,0 +1,919 @@
+use crate::event::metric::{Metric, MetricKind, MetricValue};
+use chrono::{DateTime, Utc};
+use lazy_static::lazy_static;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt;
+
+lazy_static! {
+    static ref SCOREBOARD: HashMap<char, &'static str> = vec![
+        ('_', "waiting"),
+        ('S', "starting"),
+        ('R', "reading"),
+        ('W', "sending"),
+        ('K', "keepalive"),
+        ('D', "dnslookup"),
+        ('C', "closing"),
+        ('L', "logging"),
+        ('G', "finishing"),
+        ('I', "idle_cleanup"),
+        ('.', "open"),
+    ]
+    .into_iter()
+    .collect();
+}
+
+/// enum of mod_status fields we care about
+enum StatusFieldStatistic<'a> {
+    ServerUptimeSeconds(f64),
+    TotalAccesses(f64),
+    TotalKBytes(f64),
+    TotalDuration(f64),
+    CPUUser(f64),
+    CPUSystem(f64),
+    CPUChildrenUser(f64),
+    CPUChildrenSystem(f64),
+    CPULoad(f64),
+    IdleWorkers(f64),
+    BusyWorkers(f64),
+    ConnsTotal(f64),
+    ConnsAsyncWriting(f64),
+    ConnsAsyncKeepAlive(f64),
+    ConnsAsyncClosing(f64),
+    Scoreboard(&'a str),
+}
+
+impl<'a> StatusFieldStatistic<'a> {
+    fn from_key_value(
+        key: &str,
+        value: &'a str,
+    ) -> Option<Result<StatusFieldStatistic<'a>, ParseError>> {
+        match key {
+            "ServerUptimeSeconds" => match value.parse::<f64>() {
+                Ok(value) => Some(Ok(StatusFieldStatistic::ServerUptimeSeconds(value))),
+                Err(err) => Some(Err(ParseError {
+                    key: key.to_string(),
+                    err: err.into(),
+                })),
+            },
```

Comment:
This feels cleaner, thanks for the suggestion!

The compiler did walk me into quite the type signature:

```rust
fn parse_value<T: std::str::FromStr>(key: &str, value: &str) -> Result<T, ParseError>                     
where                                                                                                     
    <T as std::str::FromStr>::Err: std::error::Error,                                                     
    <T as std::str::FromStr>::Err: 'static,                                                               
{                                                                                                         
    value.parse::<T>().map_err(|err| ParseError {                                                         
        key: key.to_string(),                                                                             
        err: err.into(),                                                                                  
    })                                                                                                    
}  
```

Reply:
Even though the compiler recommends it, I think you can simplify the `where` clause since you already have `T: FromStr`:
```rust
fn parse_value<T>(key: &str, value: &str) -> Result<T, ParseError>
where
    T: FromStr,
    T::Err: std::error::Error + 'static
```